lib/worker/src/domain/vault.rs

```diff
@@ -0,0 +1,30 @@
+use regex::Regex;
+
+#[derive(Debug)]
+pub struct Vault {
+    ns: String,
+    rel: String,
+}
+
+impl Vault {
+    pub fn from(s: String) -> Result<Vault, String> {
+        let re = Regex::new(r"^([a-zA-Z_][a-zA-Z0-9_]*)[.]([a-zA-Z_][a-zA-Z0-9_]*$)").unwrap();
```

Comment:
validates the string. again, not sure this pattern will still hold in the future, but keeping it for now